homeassistant/components/tesla_fleet/button.py

```diff
@@ -0,0 +1,90 @@
+"""Button platform for Tesla Fleet integration."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import Any
+
+from tesla_fleet_api.const import Scope
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TeslaFleetConfigEntry
+from .entity import TeslaFleetVehicleEntity
+from .helpers import handle_vehicle_command
+from .models import TeslaFleetVehicleData
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class TeslaFleetButtonEntityDescription(ButtonEntityDescription):
+    """Describes a TeslaFleet Button entity."""
+
+    func: Callable[[TeslaFleetButtonEntity], Awaitable[Any]] | None = None
+
+
+DESCRIPTIONS: tuple[TeslaFleetButtonEntityDescription, ...] = (
+    TeslaFleetButtonEntityDescription(key="wake"),  # Every button runs wakeup
```

Comment:
If you make a lambda here which isn't doing anything we can skip the if-guard in the entity and also type `func` without `None`

Reply:
Done, but I couldnt find a better awaitable that does nothing than `asyncio.sleep`. If you have a better suggestion let me know.

Reply:
Make one
```diff
diff --git a/homeassistant/components/tesla_fleet/button.py b/homeassistant/components/tesla_fleet/button.py
index 2dfb47db80..548bf06539 100644
--- a/homeassistant/components/tesla_fleet/button.py
+++ b/homeassistant/components/tesla_fleet/button.py
@@ -2,7 +2,6 @@
 
 from __future__ import annotations
 
-from asyncio import sleep
 from collections.abc import Awaitable, Callable
 from dataclasses import dataclass
 from typing import Any
@@ -21,6 +20,10 @@ from .models import TeslaFleetVehicleData
 PARALLEL_UPDATES = 0
 
 
+async def do_nothing() -> None:
+    """Do nothing."""
+
+
 @dataclass(frozen=True, kw_only=True)
 class TeslaFleetButtonEntityDescription(ButtonEntityDescription):
     """Describes a TeslaFleet Button entity."""
@@ -30,7 +33,7 @@ class TeslaFleetButtonEntityDescription(ButtonEntityDescription):
 
 DESCRIPTIONS: tuple[TeslaFleetButtonEntityDescription, ...] = (
     TeslaFleetButtonEntityDescription(
-        key="wake", func=lambda self: sleep(0)
+        key="wake", func=lambda self: do_nothing()
     ),  # Every button runs wakeup, so func does nothing
     TeslaFleetButtonEntityDescription(
         key="flash_lights", func=lambda self: self.api.flash_lights()
~
```

Reply:
Good idea. Done